packages/block-library/src/text-columns/edit.js

```diff
@@ -64,7 +59,7 @@ export default function TextColumnsEdit( { attributes, setAttributes } ) {
 						>
 							<RichText
 								tagName="p"
-								value={ get( content, [ index, 'children' ] ) }
+								value={ content?.[ index ]?.children }
```

Comment:
We're being flexible here since the `content` comes from an outside attribute and could potentially not exist.